src/DataStructures/Variables.hpp

```diff
@@ -398,3 +399,44 @@ bool operator!=(const Variables<TagsList>& lhs,
                 const Variables<TagsList>& rhs) {
   return not(lhs == rhs);
 }
+
+namespace MakeWithValueImpl_detail {
+/// \ingroup DataStructures
+/// \brief Returns a Variables whose DataVectors are the same size as `input`,
+/// with each element equal to `value`.
+template <typename... Tags>
```

Comment:
You don't need tags, so just do `template <typename TagList>`. Apply liberally to rest of file :)